### PR TITLE
Fixes resource generation when using msbuild under mono.

### PR DESCRIPTION
**Pull request by nosami**
See https://bugzilla.xamarin.com/show_bug.cgi?id=53195

### CODE REVIEW

File: src/fsharp/FSharp.Build/Microsoft.FSharp.targets [hunk 1]
```diff
@@ -66,7 +66,7 @@ this file.
 
     <PropertyGroup>
         <UsingXBuild>false</UsingXBuild>
-        <UsingXBuild Condition="$(MSBuildExtensionsPath32.Contains('xbuild'))">true</UsingXBuild>
+        <UsingXBuild Condition="'$(MSBuildAssemblyVersion)' == ''">true</UsingXBuild>
```
Comment (KevinRansom) [inline src/fsharp/FSharp.Build/Microsoft.FSharp.targets]: @nosami 
Is that really the best indicator that we are building under xbuild?

Comment (nosami) [inline src/fsharp/FSharp.Build/Microsoft.FSharp.targets]: This fix came from @radical who is the msbuild specialist for Xamarin.

Comment (nosami) [inline src/fsharp/FSharp.Build/Microsoft.FSharp.targets]: Mono on OSX already contains this patch.

Comment (KevinRansom) [inline src/fsharp/FSharp.Build/Microsoft.FSharp.targets]: Well ... that would seem to indicate it is then :-).

Thanks

Comment (nosami) [inline src/fsharp/FSharp.Build/Microsoft.FSharp.targets]: Cheers!

Comment (radical) [inline src/fsharp/FSharp.Build/Microsoft.FSharp.targets]: Just for info sake, if you want to differentiate between `msbuild` running on mono/.net/netcore, then you can use `$(MSBuildRuntimeType)` which can have values `Mono`, `Full` or `Core`.

Comment (dsyme) [inline src/fsharp/FSharp.Build/Microsoft.FSharp.targets]: @radical - what will XBuild set  ``$(MSBuildRuntimeType)`` to? thanks

Comment (radical) [inline src/fsharp/FSharp.Build/Microsoft.FSharp.targets]: It doesn't set it at all.

**Pull request merged**